src/datapattern.jl

```diff
@@ -6,7 +6,7 @@ LearnBase.getobs(dt::AbstractDataFrame, idx) = dt[idx,:]
 
 LearnBase.nobs(dt::DataFrameRow) = 1  # it is a observation
 function LearnBase.getobs(dt::DataFrameRow, idx)
-    idx == 1:1 || throw(ArgumentError(
+    only(idx) == 1 || throw(ArgumentError(
```

Comment:
I understand that `idx` can be any collection as long as it has a single element equal to 1. Right?